File: sdks/python/examples/market-data/execution_price.py
import pmxt

def main():
    api = pmxt.Polymarket()
    markets = api.search_markets('Trump')
    outcome_id = markets[0].outcomes[0].id

    order_book = api.fetch_order_book(outcome_id)
    price = api.get_execution_price(order_book, 'buy', 100)
    print(f"Average price for 100 shares: {price}")

    # Get detailed information
    detailed = api.get_execution_price_detailed(order_book, 'buy', 100)
    print(f"Filled: {detailed.filled_amount}, Fully filled: {detailed.fully_filled}")

if __name__ == "__main__":
    main()

File: sdks/typescript/examples/market-data/execution_price.ts
import pmxt from 'pmxtjs';

const main = async () => {
    const api = new pmxt.Polymarket();
    const markets = await api.searchMarkets('Trump');
    const outcomeId = markets[0].outcomes[0].id;

    const orderBook = await api.fetchOrderBook(outcomeId);
    const price = await api.getExecutionPrice(orderBook, 'buy', 100);
    console.log(`Average price for 100 shares: ${price}`);

    // Get detailed information
    const detailed = await api.getExecutionPriceDetailed(orderBook, 'buy', 100);
    console.log(`Filled: ${detailed.filledAmount}, Fully filled: ${detailed.fullyFilled}`);
};

main();
